### PR TITLE
Issue #196: Add the 'name' to the 'reference' block for FOREIGN KEY constraints

### PR DESCRIPTION
**Pull request by cfhowes**
As requested in #196, added `name` to the `references` block of constraints.  This should allow users of the parsed output to be able to associate the FOREIGN KEY with the proper column in the table.

### CODE REVIEW

File: simple_ddl_parser/dialects/sql.py [hunk 1]
```diff
@@ -960,8 +967,6 @@ def set_constraint(
             target_dict["constraints"] = {}
         if not target_dict["constraints"].get(_type):
             target_dict["constraints"][_type] = []
-        if "name" in constraint:
```
Comment (cfhowes) [inline simple_ddl_parser/dialects/sql.py:963]: i don't know why these were here, but they were breaking my references.  Removing them did not break any other tests, so i think that maybe they were unnecessary?

File: tests/dialects/test_snowflake.py [hunk 2]
```diff
@@ -644,7 +644,9 @@ def test_double_single_quotes():
 
 def test_autoincrement_order():
     # test for https://github.com/xnuinside/simple-ddl-parser/issues/208
-    ddl = """CREATE TABLE table (surrogatekey_SK NUMBER(38,0) NOT NULL autoincrement start 1 increment 1 ORDER COMMENT 'Record Identification Number Ordered')"""
+    ddl = """CREATE TABLE table (
```
Comment (cfhowes) [inline tests/dialects/test_snowflake.py:647]: they changes in this file are all from the pre-commit hook.

File: simple_ddl_parser/dialects/sql.py (hunk 1 above)
Comment (xnuinside) [reply simple_ddl_parser/dialects/sql.py:963]: possible, it is redunant code, looks like I changed some logic, but forgot to remove this

**Review by xnuinside — APPROVED** [review]

**Comment by xnuinside** [comment]
@cfhowes thanks for the PR again ! :)

**Comment by xnuinside** [comment]
@cfhowes changes released in version 0.32.0

**Pull request merged**